### ISSUE TITLE
Issue with whitespaces and parentheses when using NFS on Mac OS

### ISSUE THREAD

**Issue by matism** [hashicorp/vagrant#11439]
Please note that the Vagrant issue tracker is in priority reserved for bug reports and enhancements. For general usage questions, please use
HashiCorp Discuss: https://discuss.hashicorp.com/c/vagrant/24 Thank you!

**Tip:** Before submitting your issue, don't hesitate to remove the above introductory text, possible empty sections (e.g. References), and this tip.

### Vagrant version
2.2.7

### Host operating system
Mac OS X 10.15.3

### Guest operating system
Ubuntu 18.04.3 LTS (GNU/Linux 4.15.0-74-generic x86_64)

### Vagrantfile
```ruby
  config.vm.synced_folder "/System/Volumes/Data/Users/USERNAME/Dropbox (personal)/Sites", "/home/vagrant/Code", type: "nfs"
```

### Expected behavior
- That the folder in my dropbox gets mounted to the vagrant machines folder /home/vagrant/Code. 

- /etc/exports should contain the path within quotes or escaped:

```
"/System/Volumes/Data/Users/USERNAME/Dropbox (personal)/Sites" 192.168.10.10 -alldirs -mapall=501:20

OR

/System/Volumes/Data/Users/USERNAME/Dropbox\ \(personal\)/Sites 192.168.10.10 -alldirs -mapall=501:20
```

- "nfsd checkexports" should not return an error.

### Actual behavior
- The folder does not get mounted and vagrant up returns the following error:

```
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp,actimeo=1,nolock 192.168.10.1:/System/Volumes/Data/Users/USERNAME/Dropbox\ \(personal\)/Sites /home/vagrant/Code

Stdout from the command:



Stderr from the command:

mount.nfs: access denied by server while mounting 192.168.10.1:/System/Volumes/Data/Users/USERNAME/Dropbox\ \(personal\)/Sites
```

- /etc/exports contains the path without quotes and unescaped:

```
/System/Volumes/Data/Users/USERNAME/Dropbox (personal)/Sites -alldirs -mapall=501:20 192.168.10.10
```

- "nfsd checkexports" does return an error:

```
can't get physical path for /System/Volumes/Data/Users/USERNAME/Dropbox
Can't allocate memory to add export path: /System/Volumes/Data/Users/USERNAME/Dropbox
can't get physical path for /System/Volumes/Data/Users/USERNAME/Dropbox
exports:2: Can't allocate memory to add directory: /System/Volumes/Data/Users/USERNAME/Dropbox
```
I tried everything from escaping the path within my Vagrantfile to using symlinks without spaces or special chars but wasn't able to fix that issue. So I tried downgrading to vargant 2.0.1 and fortunately it works fine:

- /etc/exports contains the path within quotes.
- "nfsd checkexports" does not return an error.
- The folder gets mounted to my vagrant machine.

**Comment by jbonhag** [comment]
Hi @matism, thanks for bringing this to our attention! This should be fixed in the next Vagrant release.

Cheers! 😄

**Comment by matism** [comment]
@jbonhag That fast quick-thanks a lot! 👍

**Comment by matism** [comment]
@jbonhag do you have any idea when the next version will be released or is there a way to use 2.2.7 with this changes? Thanks
